docs/dev/versions.md

```diff
@@ -0,0 +1,56 @@
+# Cirq version compatibility
+
+## Overview
+
+This document defines backwards compatibility guidelines across different versions of Cirq (either for code or data).
+
+
+## Semantic Versioning 2.0
+
+Cirq follows Semantic Versioning 2.0 ([semver](http://semver.org/)) for its public API. Each release version of Cirq has the form MAJOR.MINOR.PATCH. Changes to each number have the following meaning:
+
+
```

Comment:
nit: why so many newlines between paragraphs in the markdown?

Reply:
Because I used a plugin to generate markdown from my google doc :)

Fixed!